crates/task-impls/src/consensus.rs

```diff
@@ -727,6 +760,9 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>, A: ConsensusApi<TYPES, I> +
                                         .last_synced_block_height
                                         .set(usize::try_from(leaf.get_height()).unwrap_or(0));
                                 }
+                                if let Some(ref upgrade_cert) = proposal.data.upgrade_certificate {
+                                    self.decided_upgrade_cert = Some(upgrade_cert.clone());
+                                }
```

Comment:
Nit: I think we should at least log this event at info level with the details of the upgrade cert

Reply:
Oh, yes -- this is definitely worth logging. Added!